File: martianrun/core/src/com/gamestudio24/martianrun/actors/menu/LeaderboardButton.java
package com.gamestudio24.martianrun.actors.menu;

import com.badlogic.gdx.math.Rectangle;
import com.gamestudio24.martianrun.enums.GameState;
import com.gamestudio24.martianrun.utils.Constants;
import com.gamestudio24.martianrun.utils.GameManager;

public class LeaderboardButton extends GameButton {

    public interface LeaderboardButtonListener {
        void onLeaderboard();
    }

    private final LeaderboardButtonListener listener;

    public LeaderboardButton(Rectangle bounds, LeaderboardButtonListener listener) {
        super(bounds);
        this.listener = listener;
    }

    @Override
    protected String getRegionName() {
        return Constants.LEADERBOARD_REGION_NAME;
    }

    @Override
    public void act(float delta) {
        super.act(delta);
        if (GameManager.getInstance().getGameState() != GameState.OVER) {
            remove();
        }
    }

    @Override
    public void touched() {
        listener.onLeaderboard();
    }

}

File: martianrun/core/src/com/gamestudio24/martianrun/actors/menu/LeaderboardButton.kt
package com.gamestudio24.martianrun.actors.menu

import com.badlogic.gdx.math.Rectangle
import com.gamestudio24.martianrun.enums.GameState
import com.gamestudio24.martianrun.utils.Constants
import com.gamestudio24.martianrun.utils.GameManager

class LeaderboardButton(bounds: Rectangle, private val listener: LeaderboardButtonListener) : GameButton(bounds) {
    interface LeaderboardButtonListener {
        fun onLeaderboard()
    }

    override fun getRegionName(): String? {
        return Constants.LEADERBOARD_REGION_NAME
    }

    override fun act(delta: Float) {
        super.act(delta)
        if (GameManager.instance.gameState != GameState.OVER) {
            remove()
        }
    }

    override fun touched() {
        listener.onLeaderboard()
    }
}
